Documentation suggestion: Explicitly specify minimum Rust/Cargo versions

I just installed atuin on Ubuntu 20.04 using the `cargo install` method and ran in to some friction on my first attempt because I was using the Ubuntu packaged version of Rust, 1.65 (for whatever reason the cargo package is labeled 0.66 but `cargo -V` reported 1.65 same as `rustc -V`).

With the Ubuntu packaged Rust `cargo install atuin` would fail at the final compilation stage with the error `error[E0658]: use of unstable library feature 'int_log'`

I came here looking to see if others have had the same problem and found #934.  That issue wasn't directly relevant to mine but the error I was getting did seem to fit with the idea that a newer version might fix it, so I figured it was worth a shot.  I removed the Ubuntu packages and installed the current stable release from rustup, which got me version 1.69, and it worked as expected.

The documentation does recommend the use of rustup, but doesn't make it clear that there is a minimum version required which is new enough that being "up to date" through a mainstream package manager might not be sufficient.  I would suggest explicitly listing a minimum supported version in the install documentation and maybe adding a note that popular operating systems package older versions.

--- a/docs/docs/guide/installation.md
+++ b/docs/docs/guide/installation.md
@@ -52,7 +52,11 @@ If you don't wish to use the installer, the manual installation steps are as fol
 
 === "Cargo"
 
-    It's best to use [rustup](https://rustup.rs/) to set up a Rust
+    Atuin requires a recent Rust toolchain. The minimum supported version is
+    set by `rust-version` in the workspace
+    [`Cargo.toml`](https://github.com/atuinsh/atuin/blob/main/Cargo.toml)
+    (currently {{ msrv }}). Distribution package managers often ship an older Rust,
+    so it's best to use [rustup](https://rustup.rs/) to set up an up-to-date
     toolchain, then you can run:
 
     ```shell
@@ -144,7 +148,10 @@ If you don't wish to use the installer, the manual installation steps are as fol
 === "Source"
 
     Atuin builds on the latest stable version of Rust, and we make no
-    promises regarding older versions. We recommend using [rustup](https://rustup.rs/).
+    promises regarding older versions. The minimum supported version is set by
+    `rust-version` in the workspace
+    [`Cargo.toml`](https://github.com/atuinsh/atuin/blob/main/Cargo.toml)
+    (currently {{ msrv }}). We recommend using [rustup](https://rustup.rs/).
 
     ```shell
     git clone https://github.com/atuinsh/atuin.git

--- a/docs/main.py
+++ b/docs/main.py
@@ -1,0 +1,22 @@
+"""mkdocs-macros hooks exposing repo facts to the docs.
+
+Keeps version numbers quoted in the docs in sync with their source of truth in
+the repo, so they can't drift. Currently exposes `{{ msrv }}`, read from the
+`channel` pinned in `rust-toolchain.toml`.
+"""
+
+import tomllib
+from pathlib import Path
+
+# The docs project lives in `docs/`; the repo root is its parent.
+_REPO_ROOT = Path(__file__).resolve().parent.parent
+_RUST_TOOLCHAIN = _REPO_ROOT / "rust-toolchain.toml"
+
+
+def _read_msrv() -> str:
+    with _RUST_TOOLCHAIN.open("rb") as f:
+        return tomllib.load(f)["toolchain"]["channel"]
+
+
+def define_env(env):
+    env.variables["msrv"] = _read_msrv()

--- a/docs/mkdocs.yml
+++ b/docs/mkdocs.yml
@@ -60,6 +60,7 @@ theme:
 
 plugins:
   - search
+  - macros # Read repo facts (e.g. the MSRV) into pages; see main.py
   - privacy # Download external assets at compile time for user privacy
   - git-revision-date-localized:
       enable_creation_date: true

--- a/docs/pyproject.toml
+++ b/docs/pyproject.toml
@@ -12,6 +12,7 @@ dependencies = [
   "mkdocs-minify-plugin>=0.8",
   "mkdocs-git-committers-plugin-2>=2.5",
   "mike>=2.1",
+  "mkdocs-macros-plugin>=1.0",
 ]
 
 [tool.uv]

